pkg/kapis/servicemesh/metrics/v1alpha2/handler.go

```diff
@@ -43,17 +86,104 @@ func getWorkloadMetrics(request *restful.Request, response *restful.Response) {
 		request.Request.URL.RawQuery = fmt.Sprintf("%s&namespaces=%s&workload=%s", request.Request.URL.RawQuery, namespace, workload)
 	}
 
-	handlers.WorkloadMetrics(request, response)
+	if len(KialiQueryUrl) == 0 {
+		handlers.WorkloadMetrics(request, response)
+	} else {
+		url := fmt.Sprintf("%s/kiali/api/namespaces/%s/workloads/%s/metrics?%s", KialiQueryUrl, namespace, workload, request.Request.URL.RawQuery)
+		resp, err := http.Get(url)
+		klog.V(4).Infof("Proxy workload metrics request to %s", url)
+
+		if err != nil {
```

Comment:
Don't Repeat Yourself, there are lots of code are exactly the same.

Reply:
fixed